[GTK] Remove the X11 renderer

carlosgcampos opened the pull request:
#### 8b0ce3683045d93cd365e91a4299f7e9babfc126
<pre>
[GTK] Remove the X11 renderer
<a href="https://bugs.webkit.org/show_bug.cgi?id=265308">https://bugs.webkit.org/show_bug.cgi?id=265308</a>

Reviewed by Adrian Perez de Castro.

The DMA-BUF renderer already falls back to shared memory if DMA-BUF is
not available, which should give similar (or even better) performance.
This patch also adds a new env var WEBKIT_DMABUF_RENDERER_FORCE_SHM to
easily test the shared memory code path.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/Cairo.cmake:
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoX11.cpp: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoX11.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
* Source/WebCore/platform/graphics/egl/GLContext.h:
* Source/WebCore/platform/graphics/egl/GLContextX11.cpp: Removed.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::supportsXComposite const): Deleted.
(WebCore::PlatformDisplayX11::supportsXDamage const): Deleted.
(WebCore::PlatformDisplayX11::visual const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebCore/platform/graphics/x11/XUniquePtr.h: Removed.
* Source/WebCore/platform/graphics/x11/XUniqueResource.cpp: Removed.
* Source/WebCore/platform/graphics/x11/XUniqueResource.h: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp:
(WebKit::BackingStore::createBackend):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.cpp:
(WebKit::AcceleratedBackingStore::checkRequirements):
(WebKit::AcceleratedBackingStore::create):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::rendererBufferMode):
(WebKit::AcceleratedBackingStoreDMABuf::BufferSHM::BufferSHM):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.cpp: Removed.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.h: Removed.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceX11.cpp: Removed.
* Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceX11.h: Removed.
* Source/WebKit/WebProcess/gtk/WebProcessMainGtk.cpp:
* Source/cmake/OptionsGTK.cmake:

Canonical link: <a href="https://commits.webkit.org/271134@main">https://commits.webkit.org/271134@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e7debef9bd25080b18d7aa2333496b4841455af

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27322 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5961 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28570 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29545 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25004 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27790 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7866 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3360 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24817 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27584 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4758 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23448 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4143 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4279 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24445 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/30184 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/23780 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24957 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24862 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30432 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/26502 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4307 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2451 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28385 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5772 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/33958 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4764 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7350 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3547 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4688 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
Always nice to see code being removed 💣 

r=me

carlosgcampos committed:
[GTK] Remove the X11 renderer
https://bugs.webkit.org/show_bug.cgi?id=265308

Reviewed by Adrian Perez de Castro.

The DMA-BUF renderer already falls back to shared memory if DMA-BUF is
not available, which should give similar (or even better) performance.
This patch also adds a new env var WEBKIT_DMABUF_RENDERER_FORCE_SHM to
easily test the shared memory code path.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/Cairo.cmake:
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoX11.cpp: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoX11.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig):
(WebCore::GLContext::createWindowContext):
(WebCore::GLContext::createOffscreen):
(WebCore::GLContext::createSharing):
* Source/WebCore/platform/graphics/egl/GLContext.h:
* Source/WebCore/platform/graphics/egl/GLContextX11.cpp: Removed.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::supportsXComposite const): Deleted.
(WebCore::PlatformDisplayX11::supportsXDamage const): Deleted.
(WebCore::PlatformDisplayX11::visual const): Deleted.
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:
* Source/WebCore/platform/graphics/x11/XUniquePtr.h: Removed.
* Source/WebCore/platform/graphics/x11/XUniqueResource.cpp: Removed.
* Source/WebCore/platform/graphics/x11/XUniqueResource.h: Removed.
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp:
(WebKit::BackingStore::createBackend):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStore.cpp:
(WebKit::AcceleratedBackingStore::checkRequirements):
(WebKit::AcceleratedBackingStore::create):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::rendererBufferMode):
(WebKit::AcceleratedBackingStoreDMABuf::BufferSHM::BufferSHM):
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.cpp: Removed.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreX11.h: Removed.
* Source/WebKit/WebProcess/WebPage/AcceleratedSurface.cpp:
(WebKit::AcceleratedSurface::create):
* Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceX11.cpp: Removed.
* Source/WebKit/WebProcess/WebPage/gtk/AcceleratedSurfaceX11.h: Removed.
* Source/WebKit/WebProcess/gtk/WebProcessMainGtk.cpp:
* Source/cmake/OptionsGTK.cmake:

Canonical link: https://commits.webkit.org/271134@main
